mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -52,10 +58,13 @@ sealed trait Matrix extends Serializable {
   /** Get a deep copy of the matrix. */
   def copy: Matrix
 
+  /** Transpose the Matrix */
```

Comment:
It returns a new matrix instance representing the transposed matrix. We should also mention that they share the same underlying data arrays.